tests/negative/filesystems/symlinks.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filesystems
+
+import (
+	"github.com/coreos/ignition/tests/register"
+	"github.com/coreos/ignition/tests/types"
+)
+
+func init() {
+	register.Register(register.NegativeTest, UsesSymlinks())
+}
+
+func UsesSymlinks() types.Test {
+	name := "No Device"
+	in := types.GetBaseDisk()
+	in[0].Partitions.AddLinks("ROOT", []types.Link{
+		{
+			Node: types.Node{
+				Name:      "tmp0",
+				Directory: "/",
+			},
+			Hard:   false,
+			Target: "/",
+		},
+	})
+	out := in
+	config := `{
+		"ignition": {"version": "$version"},
+		"storage": {
+			"filesystems": [{
+				"format": "ext4",
+				"path": "/tmp0",
+				"device": "/thistestdoesntneedthis"
```

Comment:
Should we use an actual device reference so that we ensure the test causes a legitimate failure?

Reply:
I'm not too familiar with the golang test framework, though for negative tests like this, I find it much more convincing to look for the error message we were expecting. Otherwise you can end up masking a regression.

Reply:
Since we're blackbox testing we'd have to parse logs, which I'm not super keen on. I agree it's not ideal.

Reply:
@jlebon: currently the test runner is doing a naive any failure is expected for negative testing; I'll file an issue for improving it to accept specific failure expectations.